src/storage-client/src/controller.rs

```diff
@@ -1288,6 +1288,55 @@ where
         for (id, description) in to_create {
             match description.data_source {
                 DataSource::Ingestion(ingestion) => {
+                    let remap_id = ingestion.remap_collection_id;
```

Comment:
I think this will work, but might install the read hold for the "main" collection twice.

I have a proposed refactor on top of this, that also adds quite some commentary: https://github.com/MaterializeInc/materialize/pull/17756

Reply:
😮‍💨 🌴 –– you are right about the double-install of the main read hold. Ty for the very thorough, well-documented patch. I am going to take it wholesale into this PR.